osquery/main/shell.cpp

```diff
@@ -17,15 +19,27 @@
 #include <osquery/filesystem.h>
 #include <osquery/logger.h>
 
-const std::string kShellTemp = "/tmp/osquery";
-
 int main(int argc, char *argv[]) {
+  // Get the caller's home dir for temporary storage/state management.
+  auto user = getpwuid(getuid());
+  std::string homedir;
+  if (getenv("HOME") != nullptr) {
+    homedir = std::string(getenv("HOME")) + "/.osquery";
+  } else if (user != nullptr || user->pw_dir != nullptr) {
+    homedir = std::string(user->pw_dir) + "/.osquery";
+  } else {
+    homedir = "/tmp/osquery";
+  }
+
   // The shell is transient, rewrite config-loaded paths.
-  if (osquery::pathExists(kShellTemp).ok() ||
-      boost::filesystem::create_directory(kShellTemp)) {
-    osquery::FLAGS_database_path = kShellTemp + "/shell.db";
-    osquery::FLAGS_extensions_socket = kShellTemp + "/shell.em";
-    osquery::FLAGS_disable_logging = true;
+  osquery::FLAGS_disable_logging = true;
+  if (osquery::pathExists(homedir).ok() ||
+      boost::filesystem::create_directory(homedir)) {
+    osquery::FLAGS_database_path = homedir + "/shell.db";
+    osquery::FLAGS_extensions_socket = homedir + "/shell.em";
+  } else {
+    LOG(ERROR) << "Cannot create osquery work directory: " << homedir;
+    return 1;
   }
```

Comment:
Do you think it's worth doing this in `runner` if it's called with `osquery::OSQUERY_TOOL_SHELL`?